staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3186,6 +3186,29 @@ type PodSpec struct {
 	// Default to false.
 	// +optional
 	SetHostnameAsFQDN *bool `json:"setHostnameAsFQDN,omitempty" protobuf:"varint,35,opt,name=setHostnameAsFQDN"`
+	// Specifies the OS on which pod intends to run. This field is needed during the API server admission time to
+	// distinguish among various node OSes in a heterogeneous Kubernetes cluster.
+	// +optional
+	OS *PodOS `json:"os,omitempty" protobuf:"bytes,36,opt,name=os"`
+}
+
+// OSName is the set of OS'es that can be used in OS.
+type OSName string
+
+// These are valid values for OSName
+const (
+	Linux   OSName = "linux"
+	Windows OSName = "windows"
+)
+
+// PodOS defines the OS parameters of a pod.
+type PodOS struct {
+	// Name is the name of the operating system. The supported values are linux and windows for now.
+	// Additional value may be defined in future and can be and one of:
+	// https://github.com/opencontainers/runtime-spec/blob/master/config.md#platform-specific-configuration
+	// Clients should expect to handle additional values and treat unrecognized values in this field as "",
+	// +optional
+	Name OSName `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
```

Comment:
this field is not optional... if they specify the OS struct, name must be provided